server/libmain/libmain.go

```diff
@@ -430,18 +430,13 @@ func StartAndRunServices(env environment.Env) {
 		mux.Handle("/webhooks/workflow/", httpfilters.WrapExternalHandler(env, wfs))
 	}
 
-	handler := http.Handler(mux)
-	if env.GetConfigurator().GetGRPCOverHTTPPortEnabled() {
-		handler = httpfilters.ServeGRPCOverHTTPPort(grpcServer, mux)
-	}
-
 	if sp := env.GetSplashPrinter(); sp != nil {
 		sp.PrintSplashScreen(*port, *GRPCPort)
 	}
 
 	server := &http.Server{
 		Addr:    fmt.Sprintf("%s:%d", *listen, *port),
-		Handler: handler,
+		Handler: grpc_server.ServeGRPCOverHTTPPort(grpcServer, mux),
```

Comment:
I think this is minor, but this change obscures the fact that `ServeGRPCOverHTTPPort` may or may not serve gRPC over the HTTP server. I think it would be fine to define the flag in this file so that the disable/enable behavior is more obvious.

Reply:
IMO, the desire to move this flag out of grpc (which I totally understand) is a result of me having been a little lazy in this CL about encapsulation. To fix this, I moved all of the grpc configuration into a `Register` function and made a grpc over http mux that wraps the env mux. In order to register the grpc_server with just the env, I also put the buildbuddy server and the ssl service in the env (and added registration functions for them) and put the listen address in the env. I also moved the grpc_port and grpcs_port flags into `grpc_server.go`.